### ISSUE TITLE
[QU] Shouldn't compilation be restricted to the source tree of 'entry'?

### ISSUE THREAD

**Issue by QuantumInformation** [s-panferov/awesome-typescript-loader#348]
I have a project that compiles fine when like so:

![image](https://cloud.githubusercontent.com/assets/216566/21891720/f111c99c-d8cb-11e6-8829-a563984422d8.png)

But when adding the `/demo` folder, the loader tries to build those files too, even though they are not referenced by any tree inside entry:

![image](https://cloud.githubusercontent.com/assets/216566/21891689/d24dc056-d8cb-11e6-82b5-d2e9b11592e8.png)

I would also love to see this repo join forces with ts-loader.

**Comment by davelosert** [comment]
I am having the exact same problem. I have two directries: `src` and `e2e`. When building the src-Folder with webpack, AT tries to resolve my e2e-Files as well - even though in my webpack config, I have this:

```
[...]
resolve: {
   // helpers.root just makes it an absolute path
   modules: [helpers.root('src'), helpers.root('node_modules')]
}
```

**Comment by s-panferov** [comment]
Closed in favour of #364
